frontend/taipy-gui/src/components/Taipy/DateRange.tsx

```diff
@@ -40,40 +40,26 @@ interface DateRangeProps extends TaipyActiveProps, TaipyChangeProps {
     labelEnd?: string;
     separator?: string;
     width?: string | number;
+    timeStep?: number; // New prop to define time step rounding (e.g., 15 for 15 minutes)
 }
 
 const textFieldProps = { textField: { margin: "dense" } } as BaseDateTimePickerSlotProps<Date>;
 
-const getRangeDateTime = (
-    json: string | string[] | undefined,
-    tz: string,
-    withTime: boolean
-): [Date | null, Date | null] => {
-    let dates: string[] = [];
-    if (typeof json == "string") {
-        try {
-            dates = JSON.parse(json);
-        } catch {
-            // too bad
-        }
-    } else {
-        dates = json as string[];
-    }
-    if (Array.isArray(dates) && dates.length) {
-        return dates.map((d) => getDateTime(d, tz, withTime)) as [Date, Date];
-    }
-    return [null, null];
+// Function to round the time based on the given step
+const roundToStep = (date: Date | null, step: number | undefined): Date | null => {
```

Comment:
Should go in a dateUtils.ts so that we can reuse it in dateSelector.tsx and timeSelector.tsx ...